Foundation/FileManager.swift

```diff
@@ -209,20 +240,93 @@ open class FileManager : NSObject {
                 useDarwinPaths = false
             #endif
         }
-        
-        var urls: [URL] = []
-        
+
         for domain in domains {
             if useDarwinPaths {
                 urls.append(contentsOf: darwinURLs(for: directory, in: domain))
             } else {
                 urls.append(contentsOf: xdgURLs(for: directory, in: domain))
             }
         }
-        
+#endif
+
         return urls
     }
-    #endif
+
+#if os(Windows)
+    private class func getURL(for id: KNOWNFOLDERID) -> URL {
+      var pszPath: PWSTR?
+      _ = withUnsafePointer(to: id) { id in
+        SHGetKnownFolderPath(id, DWORD(KF_FLAG_DEFAULT.rawValue), nil, &pszPath)
+      }
+      let url: URL = URL(fileURLWithPath: String(decodingCString: pszPath!, as: UTF16.self), isDirectory: true)
+      CoTaskMemFree(pszPath)
+      return url
+    }
+
+    private func windowsURLs(for directory: SearchPathDirectory, in domain: _SearchPathDomain) -> [URL] {
+      guard domain == .user else { return [] }
```

Comment:
The Local domain can be \ProgramData, perhaps? As long as it’s a directory or set of directories all users can read from.

Reply:
`\ProgramData` is `%ALLUSERSPROFILE%`, its readable by all.

Reply:
Yeah, that serves pretty much the same function as `/Library`, or the Darwin Library in the local domain.

Now, this doesn't _have_ the same guarantees as /Library's structure so it's not appropriate to return as the path for `.libraryDirectory`, IMHO, but we can return a local domain app support path in there, for example.

Reply:
Okay, I'll sink the guard into the various cases and add the local domain for the app data